env0/data_sshkey.go

```diff
@@ -30,44 +30,66 @@ func dataSshKey() *schema.Resource {
 }
 
 func dataSshKeyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	apiClient := meta.(*client.ApiClient)
-
 	name, nameSpecified := d.GetOk("name")
 	var sshKey client.SshKey
+	var err diag.Diagnostics
 	if nameSpecified {
-		sshKeys, err := apiClient.SshKeys()
+		sshKey, err = getSshKeyByName(meta, name)
 		if err != nil {
-			return diag.Errorf("Could not query ssh keys: %v", err)
-		}
-		for _, candidate := range sshKeys {
-			if candidate.Name == name {
-				sshKey = candidate
-			}
-		}
-		if sshKey.Name == "" {
-			return diag.Errorf("Could not find an env0 ssh key with name %s", name)
+			return err
 		}
 	} else {
 		id, idSpecified := d.GetOk("id")
 		if !idSpecified {
 			return diag.Errorf("At lease one of 'id', 'name' must be specified")
 		}
-		sshKeys, err := apiClient.SshKeys()
+		sshKey, err = getSshKeyById(meta, id)
 		if err != nil {
-			return diag.Errorf("Could not query ssh keys: %v", err)
-		}
-		for _, candidate := range sshKeys {
-			if candidate.Id == id.(string) {
-				sshKey = candidate
-			}
-		}
-		if sshKey.Name == "" {
-			return diag.Errorf("Could not find an env0 ssh key with id %s", id)
+			return err
 		}
+
 	}
 
 	d.SetId(sshKey.Id)
 	d.Set("name", sshKey.Name)
 
 	return nil
 }
+
+func getSshKeyByName(name interface{}, meta interface{}) (client.SshKey, diag.Diagnostics) {
+	apiClient := meta.(*client.ApiClient)
+
+	sshKeys, err := apiClient.SshKeys()
+	var sshKey client.SshKey
+	if err != nil {
+		return client.SshKey{}, diag.Errorf("Could not query ssh keys: %v", err)
+	}
+	for _, candidate := range sshKeys {
+		if candidate.Name == name {
+			sshKey = candidate
+		}
+	}
```

Comment:
We should error if there's more than one match by name

Reply:
https://github.com/env0/terraform-provider-env0/pull/40/commits/5c7c8e1ccfb06b6ffc63481c846347fbdbbdec6e